documentation/matlab-api.md

```diff
@@ -0,0 +1,381 @@
+# MATLAB API discussion
+
+This document is intended to describe the means by which argument flags should be passed into PACE MATLAB programs.
+This is discussed in conjunction with an observation that a migration towards a Python interface is intended in the near future.
+
+## Current Use in Language
+### MATLAB
+MATLAB uses a mixture of flag and keyword value args. Examples from the standard language include, which mostly seem to revolve around interactions with the underlying system:
+```
+load(filename,'-ascii')
+whos('-file','durer.mat')
+checkcode('-id', filename)
+```
+
+Although Herbert's variant of `matlab_xunit` based `runtests` uses flags:
+```
+runtests(tests, '-verbose')
+```
+Core MATLAB seems to have made a move more towards keyword-value
+```
+arrayfun(@()(), list, 'UniformOutput', true)
+runtests(pwd,'IncludeSubfolders',true)
+runtests(tests, 'LoggingLevel', matlab.unittest.Verbosity.VERBOSE)
+```
+
+### Python
+Python's main method of handling variable arguments is to use the standard `*args` and `**kwargs` arguments.
+
+```
+def function_name(*args):
+    for arg in args:
+        print(arg)
+
+function_name(1,2,3,4)
+
+def function_name(**kwargs):
+    if kwargs.get('header', False):
+        print(header)
+        del kwargs['header']
```

Comment:
A bit nit-picky since this is only documentation, but `header` is undefined. Did you mean something like:
```python
if kwargs.get('header', False):
    print(kwargs.pop('header'))
```

Reply:
Originally I had written
```python
if header := kwargs.get('header', False):
```
But thought that would add more confusion and removed it without replacing it